miner/payload_building.go

```diff
@@ -182,47 +193,87 @@ func (payload *Payload) ResolveFull() *engine.ExecutionPayloadEnvelope {
 		// terminates the background construction process.
 		payload.cond.Wait()
 	}
-	// Terminate the background payload construction
-	select {
-	case <-payload.stop:
-	default:
-		close(payload.stop)
+
+	payload.stopBuilding()
+
+	if payload.full != nil {
+		return engine.BlockToExecutableData(payload.full, payload.fullFees, payload.sidecars)
+	} else if !onlyFull && payload.empty != nil {
+		return engine.BlockToExecutableData(payload.empty, big.NewInt(0), nil)
+	} else if err := payload.err; err != nil {
+		log.Error("Error building any payload", "id", payload.id, "err", err)
```

Comment:
The latest payload building error is now at least bubbled up to here, if no block could be built at all.